internal/operator/operator.go

```diff
@@ -1522,9 +1615,10 @@ func (r *minioUserReconciler) Reconcile(ctx context.Context, req reconcile.Reque
 			return failure(err)
 		}
 
-		l.Info("check if minio user exists")
+		l.Info("get minio user")
 		_, err = mtac.GetUserInfo(ctx, u.Spec.AccessKey)
-		if err == nil {
+		exists := err == nil
+		if exists && !u.Spec.Migrate {
```

Comment:
Errors are not raised if a user already exists (- instead this performs an update).  We augment the existing existence check to consider the `.spec.migrate` field - if its set, we do not raise our own 'user already exists' error.